### PR TITLE
docs and updates

### PR DESCRIPTION
**Pull request by kshitij12345**

### CODE REVIEW

File: src/neptune_pytorch/impl/__init__.py [hunk 1]
```diff
@@ -13,25 +13,79 @@
...
 class NeptuneLogger:
+    """Captures model training metadata and logs them to Neptune.
+
+    Note: To use this module, you need to have PyTorch installed.
+
+    Args:
+        run: Neptune run object. You can also pass a namespace handler object;
+            for example, run["test"], in which case all metadata is logged under
+            the "test" namespace inside the run.
+        base_namespace: Namespace where all metadata logged by the callback is stored.
```
Comment (kshitij12345) [inline src/neptune_pytorch/impl/__init__.py:56]: Similar to keras integration, have skipped on adding default value here

https://github.com/neptune-ai/neptune-tensorflow-keras/blob/3b3dec493dc42c9fe539cf26175819151715d12e/src/neptune_tensorflow_keras/impl/__init__.py#L79-L80

File: src/neptune_pytorch/impl/__init__.py [hunk 2]
```diff
@@ -13,25 +13,79 @@
...
 except ImportError:
     IS_TORCHVIZ_AVAILABLE = False
 
+INTEGRATION_VERSION_KEY = "source_code/integrations/neptune-pytorch"
+
 
 class NeptuneLogger:
+    """Captures model training metadata and logs them to Neptune.
+
+    Note: To use this module, you need to have PyTorch installed.
```
Comment (normandy7) [inline src/neptune_pytorch/impl/__init__.py:50]: Maybe this is not necessary, as in this case it should be clear that it's about PyTorch :)

**Review by normandy7 — APPROVED** [review]

File: src/neptune_pytorch/impl/__init__.py [hunk 3]
```diff
@@ -13,25 +13,79 @@
...
 
 from neptune_pytorch.impl.version import __version__
 
+try:
+    # neptune-client>=1.0.0 package structure
+    from neptune import Run
+    from neptune.handler import Handler
+    from neptune.internal.utils import verify_type
+except ImportError:
+    # neptune-client=0.9.0+ package structure
```
Comment (AleksanderWWW) [inline src/neptune_pytorch/impl/__init__.py:33]: Let's maybe change this comment (or remove it altogether) as it is misleading

**Comment by AleksanderWWW** [comment]
> Also about `save_checkpoint` method - don't we want to use uploading from stream?

@kshitij12345

**Comment by kshitij12345** [comment]
Thanks for the ping

In this case, we don't save a temporary copy of the model. So, we don't have the stream but we have the name.

Though, I was wondering if we should or shouldn't save a copy on local disk. Wdyt @AleksanderWWW ?

**Review by AleksanderWWW — APPROVED** [review]

File: src/neptune_pytorch/impl/__init__.py [hunk 4]
```diff
@@ -13,25 +13,81 @@
...
+        log_freq: How often to log the parameters/gradients norm. Applicable only
+            if `log_parameters` or `log_gradients` is True.
+        log_model_diagram: Whether to save the model visualization.
+            Requires torchviz to be installed: https://pypi.org/project/torchviz/
+
+    Example:
+        import neptune
+        from neptune.integrations.pytorch import NeptuneLogger
+        run = neptune.init_run()
+        neptune_callback = NeptuneLogger(run=run, model=model)
```
Comment (normandy7) [inline src/neptune_pytorch/impl/__init__.py:70]: Just noticed one thing. What's the `model` parameter here?

Comment (kshitij12345) [reply src/neptune_pytorch/impl/__init__.py:70]: Ah, this is the PyTorch `model` (of type torch.nn.Module) that will be tracked by the integration. Missed that 😓 . Will update the PR soon. 

Thank you!

Comment (kshitij12345) [reply src/neptune_pytorch/impl/__init__.py:71]: Ping @normandy7

Comment (normandy7) [reply src/neptune_pytorch/impl/__init__.py:71]: Looks good, thanks!

**Review by normandy7 — APPROVED** [review]

**Pull request merged**